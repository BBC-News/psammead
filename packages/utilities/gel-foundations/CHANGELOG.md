# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

