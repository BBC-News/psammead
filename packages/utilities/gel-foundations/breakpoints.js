module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
