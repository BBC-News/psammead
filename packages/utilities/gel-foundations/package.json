{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
