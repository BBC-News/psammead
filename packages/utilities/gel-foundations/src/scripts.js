export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
