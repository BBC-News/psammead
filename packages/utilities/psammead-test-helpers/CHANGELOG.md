# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
