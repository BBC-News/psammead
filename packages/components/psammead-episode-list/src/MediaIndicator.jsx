import React from 'react';
import styled from '@emotion/styled';
import { string, number } from 'prop-types';
import { GEL_SPACING, GEL_SPACING_DBL } from '@bbc/gel-foundations/spacings';
import {
  GEL_GROUP_1_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';

const Wrapper = styled.div`
  display: inline-block;
  width: ${props => props.size};
  height: ${props => props.size};
  @media (max-width: ${GEL_GROUP_1_SCREEN_WIDTH_MAX}) {
    margin-left: ${props => (props.dir === 'ltr' ? 0 : GEL_SPACING_DBL)};
    margin-right: ${props => (props.dir === 'ltr' ? GEL_SPACING_DBL : 0)};
    padding-top: ${props => (props.dir === 'ltr' ? 0 : '0.25rem')};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    margin-left: ${props =>
      props.dir === 'ltr' ? GEL_SPACING : GEL_SPACING_DBL};
    margin-right: ${props =>
      props.dir === 'ltr' ? GEL_SPACING_DBL : GEL_SPACING};
    padding-top: ${props => (props.dir === 'ltr' ? 0 : '0.25rem')};
  }
  vertical-align: top;
`;

const MediaIndicator = ({ size, dir }) => (
  <Wrapper
    aria-hidden="true"
    size={size}
    dir={dir}
    dangerouslySetInnerHTML={{
      __html: `
      <svg class="rounded-play-button" focusable="false" width="40" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">
        <circle class="rounded-play-button__circle" fill="none" stroke="currentColor" stroke-width="1.765" stroke-miterlimit="10" cx="20" cy="20" r="19.119"/>
        <polygon class="rounded-play-button__triangle" fill="currentColor" points="15.799,26.939 27.062,20.008 15.799,13.078 "/>
      </svg>
      `,
    }}
  />
);

MediaIndicator.propTypes = {
  size: number.isRequired,
  dir: string.isRequired,
};

export default MediaIndicator;
