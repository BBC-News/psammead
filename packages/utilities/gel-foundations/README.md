# gel-foundations &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/gel-foundations.svg)](https://www.npmjs.com/package/@bbc/gel-foundations) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

This package provides a range of string constants for use in CSS, intended to help implement [BBC GEL-compliant](https://www.bbc.co.uk/gel/articles/what-is-gel) webpages and components.

## Exports

`/breakpoints` - GEL breakpoints, as well as typography breakpoints. These use the GEL grid sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes).  
`/spacings` - GEL spacings and GEL Grid margins and gutters. These use the GEL grid spacing sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#spacing-layout).  
`/typography` - GEL typography. These are based on the gel typography standard which can be found in the [GEL guidelines](https://www.bbc.co.uk/gel/guidelines/typography). [More details on how these sizes were implemented for this package are available here.](./typography_sizes_web.md)

## Installation

```jsx
npm install @bbc/gel-foundations --save
```

## Usage

```jsx
import { GEL_GROUP_3_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';

import { GEL_GUTTER_ABOVE_600PX } from '@bbc/gel-foundations/spacings';

import { GEL_BREVIER } from '@bbc/gel-foundations/typography';
```

These values can then be used directly within CSS declarations in code:

```jsx
import { css } from 'styled-components';

const SomeStyledComponent = css`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-gap: ${GEL_GUTTER_ABOVE_600PX};
    ${GEL_BREVIER};
  }
`;
```

To allow the typography to be fully accessible and responsive, please note that you should apply a default font-size to the document root (e.g. `html { font-size: 100% }`).

Our typography uses `em` for font-size and `rem` for line-height. `em` allows modularity of components: you can change the component font-size by changing the font-size of its container. `rem` is relative to the document root, so we use that for line-height and spacing for a consistent look-and-feel across the document. You can read our [detailed analysis of "REMs vs EMs for spacing"](https://github.com/bbc/simorgh/blob/latest/docs/Spacing-Units.md) for more information.

## Contributing

When **adding** a new export to this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) also need to be updated. When **removing** an exisiting export from this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) need to be updated and the package version requires a major change (EG: 1.2.1 -> 2.0.0) as this would be considered a breaking change as functionality is being removed.

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
