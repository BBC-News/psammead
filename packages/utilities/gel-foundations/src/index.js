import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
