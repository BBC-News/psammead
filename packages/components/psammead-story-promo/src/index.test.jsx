import React, { Fragment } from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import MediaIndicator from '@bbc/psammead-media-indicator';
import { render } from '@testing-library/react';
import StoryPromo, { Headline, Summary, Link } from './index';

const Image = <img src="https://foobar.com/image.png" alt="Alt text" />;

// eslint-disable-next-line react/prop-types
const Info = ({ topStory }) => (
  <Fragment>
    <Headline script={latin} topStory={topStory}>
      <Link href="https://www.bbc.co.uk/news">The headline of the promo</Link>
    </Headline>
    <Summary script={latin} topStory={topStory}>
      The summary of the promo
    </Summary>
    <time>12 March 2019</time>
  </Fragment>
);

const mediaInfo = (
  <MediaIndicator
    duration="2:15"
    datetime="PT2M15S"
    offscreenText="Video 2 minutes 15 seconds"
  />
);

describe('StoryPromo', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo image={Image} info={Info({ topStory: false })} />,
  );
});

describe('StoryPromo with Media Indicator', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo
      image={Image}
      info={Info({ topStory: false })}
      mediaIndicator={mediaInfo}
    />,
  );
});

describe('StoryPromo - Top Story', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo image={Image} info={Info({ topStory: true })} topStory />,
  );
});

describe('StoryPromo - Top Story with Media Indicator', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo
      image={Image}
      info={Info({ topStory: true })}
      mediaIndicator={mediaInfo}
      topStory
    />,
  );
});

describe('assertions', () => {
  it('should render h3, a, p, time', () => {
    const { container, getAllByText } = render(
      <StoryPromo
        image={Image}
        info={Info({ topStory: true })}
        mediaIndicator={mediaInfo}
      />,
    );

    expect(container.querySelector('h3 a').textContent).toEqual(
      'The headline of the promo',
    );
    expect(container.querySelector('p').textContent).toEqual(
      'The summary of the promo',
    );

    expect(getAllByText('Video 2 minutes 15 seconds')[0]).toBeTruthy();
    expect(getAllByText('2:15')[0]).toBeTruthy();

    const image = container.querySelector('img');

    expect(image.getAttribute('src')).toEqual('https://foobar.com/image.png');
    expect(image.getAttribute('alt')).toEqual('Alt text');
  });
});
