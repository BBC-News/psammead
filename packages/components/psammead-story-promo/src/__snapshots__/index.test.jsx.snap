// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryPromo - Top Story should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  position: relative;
  margin-bottom: 0.5rem;
  width: 100%;
}

.c2 {
  position: relative;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c6 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c3 {
  display: inline-block;
  vertical-align: top;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    width: 33.33%;
    margin-bottom: none;
  }
}

@media (min-width:80rem) {
  .c1 {
    width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.75rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 66.67%;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    width: 66.67%;
  }
}

@supports (display:grid) {
  .c3 {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 1 / span 6;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Alt text"
        src="https://foobar.com/image.png"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <h3
      className="c4"
    >
      <a
        className="c5"
        href="https://www.bbc.co.uk/news"
      >
        The headline of the promo
      </a>
    </h3>
    <p
      className="c6"
    >
      The summary of the promo
    </p>
    <time>
      12 March 2019
    </time>
  </div>
</div>
`;

exports[`StoryPromo - Top Story with Media Indicator should render correctly 1`] = `
.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.c6 {
  vertical-align: middle;
  margin: 0 0.25rem;
  fill: #222222;
}

.c4 {
  padding: 0.5rem 0.25rem;
  background-color: #FFFFFF;
  display: inline-block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-size: 0.75rem;
  line-height: 1rem;
  color: #222222;
  height: 2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c8 {
  vertical-align: middle;
  margin: 0 0.25rem;
}

.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  position: relative;
  margin-bottom: 0.5rem;
  width: 100%;
}

.c2 {
  position: relative;
}

.c3 {
  position: absolute;
  bottom: 0;
}

.c10 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c12 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c9 {
  display: inline-block;
  vertical-align: top;
}

.c11 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c11:hover,
.c11:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11:visited {
  color: #6E6E73;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    width: 33.33%;
    margin-bottom: none;
  }
}

@media (min-width:80rem) {
  .c1 {
    width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c10 {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1.75rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c12 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c12 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    width: 66.67%;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c9 {
    width: 66.67%;
  }
}

@supports (display:grid) {
  .c9 {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 1 / span 6;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Alt text"
        src="https://foobar.com/image.png"
      />
      <div
        className="c3"
      >
        <div
          aria-hidden="true"
          className="c4"
        >
          <div
            className="c5"
          >
            <svg
              className="c6"
              focusable="false"
              height="12px"
              viewBox="0 0 32 32"
              width="12px"
            >
              <polygon
                points="3,32 29,16 3,0"
              />
            </svg>
            <span
              className="c7"
            >
              Video 2 minutes 15 seconds
            </span>
            <time
              className="c8"
              dateTime="PT2M15S"
            >
              2:15
            </time>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    <h3
      className="c10"
    >
      <a
        className="c11"
        href="https://www.bbc.co.uk/news"
      >
        The headline of the promo
      </a>
    </h3>
    <p
      className="c12"
    >
      The summary of the promo
    </p>
    <time>
      12 March 2019
    </time>
  </div>
</div>
`;

exports[`StoryPromo should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  position: relative;
  width: 33.33%;
}

.c2 {
  position: relative;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c6 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c3 {
  display: inline-block;
  vertical-align: top;
  width: 66.67%;
  padding: 0 0.5rem;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c6 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    width: 66.67%;
  }
}

@supports (display:grid) {
  .c3 {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Alt text"
        src="https://foobar.com/image.png"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <h3
      className="c4"
    >
      <a
        className="c5"
        href="https://www.bbc.co.uk/news"
      >
        The headline of the promo
      </a>
    </h3>
    <p
      className="c6"
    >
      The summary of the promo
    </p>
    <time>
      12 March 2019
    </time>
  </div>
</div>
`;

exports[`StoryPromo with Media Indicator should render correctly 1`] = `
.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.c6 {
  vertical-align: middle;
  margin: 0 0.25rem;
  fill: #222222;
}

.c4 {
  padding: 0.5rem 0.25rem;
  background-color: #FFFFFF;
  display: inline-block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-size: 0.75rem;
  line-height: 1rem;
  color: #222222;
  height: 2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c8 {
  vertical-align: middle;
  margin: 0 0.25rem;
}

.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  position: relative;
  width: 33.33%;
}

.c2 {
  position: relative;
}

.c10 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c12 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c9 {
  display: inline-block;
  vertical-align: top;
  width: 66.67%;
  padding: 0 0.5rem;
}

.c11 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c11:hover,
.c11:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11:visited {
  color: #6E6E73;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:25rem) {
  .c3 {
    position: absolute;
    bottom: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c12 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c12 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c12 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c9 {
    width: 66.67%;
  }
}

@supports (display:grid) {
  .c9 {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Alt text"
        src="https://foobar.com/image.png"
      />
      <div
        className="c3"
      >
        <div
          aria-hidden="true"
          className="c4"
        >
          <div
            className="c5"
          >
            <svg
              className="c6"
              focusable="false"
              height="12px"
              viewBox="0 0 32 32"
              width="12px"
            >
              <polygon
                points="3,32 29,16 3,0"
              />
            </svg>
            <span
              className="c7"
            >
              Video 2 minutes 15 seconds
            </span>
            <time
              className="c8"
              dateTime="PT2M15S"
            >
              2:15
            </time>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    <h3
      className="c10"
    >
      <a
        className="c11"
        href="https://www.bbc.co.uk/news"
      >
        The headline of the promo
      </a>
    </h3>
    <p
      className="c12"
    >
      The summary of the promo
    </p>
    <time>
      12 March 2019
    </time>
  </div>
</div>
`;
